### PR TITLE
fix(router): lazy loading keeps refetching modules

### LINKED ISSUE #10702: Possible router bug/unexpected behavior
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request
https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
```HeroComponent``` is being recreated with each child navigation

**Expected/desired behavior**
```HeroComponent``` shouldn't be recreated with each child navigation.

I was having the issue in my private app, so checked a plunker from angular.io and noticed the same issue there as well.

**Reproduction of the problem**
plunker was taken from the angular.io docs and just added ```console.log```s and displayed the time
http://plnkr.co/edit/ey56dehV7rwfxkneK0oW?p=preview

Here is where i got the original plunker from angular.io https://angular.io/docs/ts/latest/guide/ngmodule.html#!#lazy-load

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.5
 
* **Browser:** [Chrome 52]

### PR DESCRIPTION
**Pull request by vsavkin**
Closes #10702,  #10680

### CODE REVIEW

**Comment by robwormald** [comment]
LGTM

**Comment by vikerman** [comment]
TAP Green

File: modules/@angular/router/src/apply_redirects.ts [hunk 1]
```diff
@@ -244,10 +244,14 @@ class ApplyRedirects {
     } else if (route.loadChildren) {
       return runGuards(injector, route).mergeMap(shouldLoad => {
         if (shouldLoad) {
-          return this.configLoader.load(injector, route.loadChildren).map(r => {
-            (<any>route)._loadedConfig = r;
-            return r;
-          });
+          if ((<any>route)._loadedConfig) {
+            return of ((<any>route)._loadedConfig);
```
Comment (vicb) [inline modules/@angular/router/src/apply_redirects.ts]: have you reported to clang ?

**Comment by SaraSlm** [comment]
how can i put lazy loading in my project? please help me

**Comment by vandensd** [comment]
Running into this just tonight, wondering if the life cycle hooks will reset (i.e. onChanges followed by onInit) as if the component is loaded for the first time.

**Pull request merged**